openssl3/Dockerfile-interop

```diff
@@ -0,0 +1,135 @@
+# Multi-stage build: First the full builder image:
+
+ARG INSTALLDIR_OPENSSL=/opt/openssl32
+ARG INSTALLDIR_LIBOQS=/opt/liboqs
+
+# liboqs build type variant; maximum portability of image:
+ARG LIBOQS_BUILD_DEFINES="-DOQS_DIST_BUILD=ON"
+
+# Define the degree of parallelism when building the image; leave the number away only if you know what you are doing
+ARG MAKE_DEFINES="-j 8"
+
+ARG SIG_ALG="dilithium3"
+
+FROM ubuntu as buildopenssl
+# Take in all global args
+ARG INSTALLDIR_OPENSSL
+ARG INSTALLDIR_LIBOQS
+ARG LIBOQS_BUILD_DEFINES
+ARG MAKE_DEFINES
+ARG SIG_ALG
+
+LABEL version="1"
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt update && apt upgrade -y
+
+# Get all software packages required for builing openssl
+RUN apt install -y build-essential clang libtool make gcc ninja-build cmake libtool wget git
+
+# get current openssl sources
+RUN mkdir /optbuild && cd /optbuild && git clone --depth 1 --branch master https://github.com/openssl/openssl.git
+
+# build OpenSSL3
+WORKDIR /optbuild/openssl
+RUN LDFLAGS="-Wl,-rpath -Wl,${INSTALLDIR_OPENSSL}/lib64" ./config shared --prefix=${INSTALLDIR_OPENSSL} && \
+    make ${MAKE_DEFINES} && make install && if [ -d ${INSTALLDIR_OPENSSL}/lib64 ]; then ln -s ${INSTALLDIR_OPENSSL}/lib64 ${INSTALLDIR_OPENSSL}/lib; fi && if [ -d ${INSTALLDIR_OPENSSL}/lib ]; then ln -s ${INSTALLDIR_OPENSSL}/lib ${INSTALLDIR_OPENSSL}/lib64; fi 
```

Comment:
Is there an outside chance that both `/opt/openssl32/lib` and `/opt/openssl32/lib64` will exist as directories? If so, might want to edit the logic here so that the symlinks are not created in that case.

Reply:
> Is there an outside chance that both /opt/openssl32/lib and /opt/openssl32/lib64 will exist as directories?

No: OpenSSL creates either, not both. I still don't understand the logic which when.